blockchain/types/transaction_signing.go

```diff
@@ -78,7 +78,7 @@ func LatestSigner(config *params.ChainConfig) Signer {
 	// Be aware that it checks whether EthTxTypeCompatibleBlock is set,
 	// but doesn't check whether it is enabled on a specific block number.
 	if config.EthTxTypeCompatibleBlock != nil {
-		return NewLondonSigner(config.ChainID)
+		return NewPragueSigner(config.ChainID)
```

Comment:
I think the logic should be:

```
	if config.IsPragueForkEnabled(blockNumber) {
		return NewPragueSigner(config.ChainID)
	} else if config.IsEthTxTypeForkEnabled(blockNumber) {
		return NewLondonSigner(config.ChainID)
	}
```

Reply:
> // Be aware that it checks whether EthTxTypeCompatibleBlock is set,
// but doesn't check whether it is enabled on a specific block number.

Following the comments above, this might be better.

```
        if config.PragueCompatibleBlock != nil {
		return NewPragueSigner(config.ChainID)
	} else if config.EthTxTypeCompatibleBlock != nil {
		return NewLondonSigner(config.ChainID)
	}
```